gst/nnstreamer/tensor_common.c

```diff
@@ -616,8 +616,11 @@ gst_tensor_config_is_equal (const GstTensorConfig * c1,
   g_return_val_if_fail (c1 != NULL, FALSE);
   g_return_val_if_fail (c2 != NULL, FALSE);
 
-  /** @todo 1/2 == 2/4 Don't say they are different! */
-  if (c1->rate_n != c2->rate_n || c1->rate_d != c2->rate_d) {
+  if (c1->rate_d == 0 || c2->rate_d == 0) {
+    return FALSE;
+  }
+
+  if (c1->rate_n * c2->rate_d != c1->rate_d * c2->rate_n) {
```

Comment:
I think you can implement a helper function for those comparisons.

Reply:
Be prepared for overflows as well.

Reply:
Use https://developer.gnome.org/gstreamer/stable/gstreamer-GstUtils.html#gst-util-fraction-compare for comparison

Reply:
Updated.